manager/controllers/app/policy_compiler_interface.go

```diff
@@ -10,70 +10,74 @@ import (
 	app "fybrik.io/fybrik/manager/apis/app/v1alpha1"
 	"fybrik.io/fybrik/manager/controllers/utils"
 	connectors "fybrik.io/fybrik/pkg/connectors/clients"
-	pb "fybrik.io/fybrik/pkg/connectors/protobuf"
+	openapiclientmodels "fybrik.io/fybrik/pkg/taxonomy/model/base"
 	"fybrik.io/fybrik/pkg/vault"
+	"github.com/gdexlab/go-render/render"
 )
 
-// ConstructApplicationContext constructs ApplicationContext structure to send to Policy Compiler
-func ConstructApplicationContext(datasetID string, input *app.FybrikApplication, operation *pb.AccessOperation) *pb.ApplicationContext {
-	var credentialPath string
-	if input.Spec.SecretRef != "" {
-		credentialPath = utils.GetVaultAddress() + vault.PathForReadingKubeSecret(input.Namespace, input.Spec.SecretRef)
+func ConstructOpenAPIReq(datasetID string, input *app.FybrikApplication, operation *openapiclientmodels.PolicyManagerRequestAction) *openapiclientmodels.PolicyManagerRequest {
+	req := openapiclientmodels.PolicyManagerRequest{}
+	action := openapiclientmodels.PolicyManagerRequestAction{}
+	resource := openapiclientmodels.Resource{}
+
+	resource.SetName(datasetID)
+	req.SetResource(resource)
+
+	destination := operation.GetDestination()
+	action.SetDestination(destination)
+	operationType := operation.GetActionType()
+	if operationType == openapiclientmodels.READ {
+		action.SetActionType(openapiclientmodels.READ)
 	}
-	return &pb.ApplicationContext{
-		AppInfo: &pb.ApplicationDetails{
-			ProcessingGeography: operation.Destination,
-			Properties:          input.Spec.AppInfo,
-		},
-		CredentialPath: credentialPath,
-		Datasets: []*pb.DatasetContext{{
-			Dataset: &pb.DatasetIdentifier{
-				DatasetId: datasetID,
-			},
-			Operation: operation,
-		}},
+	if operationType == openapiclientmodels.WRITE {
+		action.SetActionType(openapiclientmodels.WRITE)
```

Comment:
Why do we need the if conditions and not just set operationType

Reply:
fixed. Thanks.